### PR TITLE
Improve docker logs, move pullingImage::awaitCompletion inside retry

### PR DESCRIPTION
**Pull request by lindlind**

### CODE REVIEW

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 1]
```diff
@@ -54,19 +55,23 @@ public DockerEnvironment(BaseEnvConfig config, DockerClient client) {
     public void install(StreamQueue.LogHandle handle) throws EnvironmentInstallationException {
         if (containerId != null) {
             handle.logOut("Using already running container from cache");
```
Comment (lindlind) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:57]: user-readable log

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 2]
```diff
@@ -54,19 +55,23 @@ public DockerEnvironment(BaseEnvConfig config, DockerClient client) {
...
+            LOG.info("Using already running container from cache; containerId: {}", containerId);
```
Comment (lindlind) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:58]: internal grep'able log

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 3]
```diff
@@ -245,18 +262,17 @@ public BaseEnvConfig config() {
         return config;
     }
 
-    private void prepareImage(BaseEnvConfig config, StreamQueue.LogHandle handle) {
-        handle.logOut("Pulling image {} ...", config.image());
-        final var pullingImage = retry.executeSupplier(() -> client
-            .pullImageCmd(config.image())
-            .exec(new PullImageResultCallback()));
-        try {
-            pullingImage.awaitCompletion();
```
Comment (lindlind) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:254]: TLS handshake timeout can be thrown from here. Move it inside retry.

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 4]
```diff
@@ -54,19 +55,23 @@ public DockerEnvironment(BaseEnvConfig config, DockerClient client) {
...
+            LOG.info("Using already running container from cache; containerId: {}", containerId);
             return;
         }
 
+        String sourceImage = config.image();
         try {
-            prepareImage(config, handle);
+            prepareImage(sourceImage, handle);
+        } catch (InterruptedException e) {
+            handle.logErr("Pulling image {} was interrupted", sourceImage);
```
Comment (imakunin) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:66]: s/Pulling image/Image pulling/

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 5]
```diff
@@ -54,19 +55,23 @@ public DockerEnvironment(BaseEnvConfig config, DockerClient client) {
...
+            throw new RuntimeException(e);
         } catch (Exception e) {
             handle.logErr("Error while pulling image: {}", e);
+            LOG.info("Error while pulling image {}", sourceImage, e);
```
Comment (imakunin) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:70]: error should be logged with ERROR level, not INFO

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 6]
```diff
@@ -63,11 +63,11 @@ public void install(StreamQueue.LogHandle handle) throws EnvironmentInstallation
         try {
             prepareImage(sourceImage, handle);
         } catch (InterruptedException e) {
-            handle.logErr("Pulling image {} was interrupted", sourceImage);
+            handle.logErr("Image pulling was interrupted");
```
Comment (Hoxo) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:66]: But why without image name?

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java [hunk 7]
```diff
@@ -245,18 +262,17 @@ public BaseEnvConfig config() {
...
+        handle.logOut("Pulling image {} ...", image);
+        AtomicInteger pullingAttempt = new AtomicInteger(0);
+        retry.executeCallable(() -> {
+            LOG.info("Pulling image {}, attempt {}", image, pullingAttempt.incrementAndGet());
+            final var pullingImage = client
+                .pullImageCmd(config.image())
+                .exec(new PullImageResultCallback());
+            return pullingImage.awaitCompletion();
+        });
+        handle.logOut("Pulling image {} done", image);
```
Comment (Hoxo) [inline lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:275]: Placing app log here will be suitable for measuring pulling time

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java (hunk 6 above)
Comment (lindlind) [reply lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:66]: It was printed for user some rows before. Originally it was the internal log and image name was helpful.

File: lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java (hunk 7 above)
Comment (lindlind) [reply lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:275]: Logs with "handle.logOut" and "handle.logErr" are also printed as app logs. (Maybe they shouldn't)

Comment (Hoxo) [reply lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java:275]: Sorry, it seems I left ambiguous definition. By 'app log' I mean usual `LOG.info(...)` and its logs which we should get in our logging UI. So I'm saying that these logs will not only be helpful for user, but for us too. For example, we'll be able to compare pulling time from dockerhub and local registry

**Pull request merged**